src/symbolic/symbolic_choice.ml

```diff
@@ -402,6 +411,7 @@ module Make (Thread : Thread_intf.S) = struct
     | `Unsat -> stop
     | `Sat -> begin
       let symbol_scopes = Symbol_scope.of_symbol symbol in
+      let* () = modify_thread Thread.mark_pending_pc_done in
```

Comment:
This should be done before calling the solver, no? Because otherwise, if the things newly added to the PC are unsat, we'll crash instead of stopping.